userland/system/applist/AppListViewController.h

```diff
@@ -45,6 +45,12 @@ class AppListViewController : public UI::ViewController<AppListView> {
         LG::PNG::PNGLoader loader;
 
         std::string icon_path = jdict_root->data()["icon_path"]->cast_to<LFoundation::Json::StringObject>()->data();
+        std::string icon_rel_path = jdict_root->data()["icon_rel_path"]->cast_to<LFoundation::Json::StringObject>()->data();
+
+        if (!icon_rel_path.empty()) {
```

Comment:
Consider adding an "empty" icon if `icon_rel_path` is not set. (here they are: https://github.com/opuntiaOS-Project/opuntiaOS/tree/master/base/res/icons/apps/missing.icon)

Reply:
Okay, so that means `icon_path` generated by `prepare_app.py` should be completely omitted? Or is there any plan for it? Maybe always setting it to the missing icon path

Reply:
We should unify `icon_path` and `icon_rel_path` for sure, so yeah we should omit one of them. But also in case if there is no such field in `info.json` which describes location of an icon, we should set a "missing icon".

As an alternative to this `icon_path` and `icon_rel_path` we could provide path to asset folder in the `info.json`. And to require have a "special" folder which contains icons inside the asset folder. In other words to have a provided path to an asset folder and look there for a "AppIcon" folder to get all of the icons.